mllib/src/test/scala/org/apache/spark/mllib/linalg/distributed/RowMatrixSuite.scala

```diff
@@ -101,6 +101,17 @@ class RowMatrixSuite extends SparkFunSuite with MLlibTestSparkContext {
     }
   }
 
+  test("getTreeAggregateIdealDepth") {
+    val nbPartitions = 10000
+    val vectors = sc.emptyRDD[Vector]
+      .repartition(nbPartitions)
+    vectors.conf.set("spark.driver.maxResultSize", "10g")
```

Comment:
So about this line: I figured out that changing this will affect the sparkConf for tests that will be executed after this one aswell (I tried to set maxResultSize to 10bytes here and it broke several later tests).
Do you knowwhat's the disired way to make this only change the sparkConf for the closure of this test ?

Reply:
Do you have to change this from 2g for it to select a depth of 3? does a little larger number of partitions help? I thought you might have to reduce it if anything. There are ways to isolate the conf change to just the test, but if it's avoidable that's better

Reply:
I checked the maxResultSize parameter value in the test and it's the default one (1gig, e.g 1024 Mb) -> I'll change the numPartitions and object size to:
- numPartitions -> 100
- objectSize -> 100 to get a desired depth of 2 (because sqrt(100) * 100 <= 1024)
- objectSize -> 110 to get a desired depth of 3 (because sqrt(100) * 110 > 1024 and math.pow(100, 0.33) * 110 > 1024)

So the (semantically) same test doesn't require a change in the sharedContext spark conf